### PR TITLE
[PT-82]-DarkMode

### PR DESCRIPTION
**Pull request by ItsFaizan**
# [PT-82](https://0xbase.atlassian.net/browse/PT-82?atlOrigin=eyJpIjoiMzJlNjMxMjdjYWY2NDU1Zjg4MjVmMzNkZTA1YTBlNGUiLCJwIjoiaiJ9) - DarkMode

### Theme updated and added dark mode in it

[PT-82]: https://0xbase.atlassian.net/browse/PT-82?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+import { darkTheme, lightTheme } from '@/utils/theme'
 
-export const metadata = {
-  title: 'Porfolio Meta',
-  description: 'One place to view all your assets',
-}
+// export const metadata = {
+//   title: 'Porfolio Meta',
+//   description: 'One place to view all your assets',
+// }
```
Comment (fuddin95) [inline src/app/layout.tsx:18]: why did you comment this?

File: src/utils/theme.ts [hunk 2]
```diff
@@ -15,3 +15,18 @@ export const theme = createTheme({
...
+    primary: {
+      main: colors.grey[50],
+    },
+    secondary: {
+      main: colors.cyan[300],
+    },
+    background: {
+      default: colors.blueGrey[900],
+    },
+    mode: 'dark',
```
Comment (fuddin95) [inline src/utils/theme.ts:30]: for now change the colours, so it is visible that the dark mode switch is working

File: src/app/layout.tsx [hunk 3]
```diff
@@ -1,26 +1,37 @@
...
   return (
     <html lang="en">
       <head></head>
       <body>
         <AppRouterCacheProvider>
-          <ThemeProvider theme={theme}>
+          <ThemeProvider theme={darkMode ? darkTheme : lightTheme}>
             <CssBaseline />
             <Navbar />
+            <Button onClick={toggleDarkMode}>{darkMode ? <Brightness7Icon /> : <Brightness4Icon />}</Button>
```
Comment (fuddin95) [inline src/app/layout.tsx:34]: Look for `switch` component in MUI it will be better instead of `Button`

File: src/app/layout.tsx (hunk 1 above)
Comment (ItsFaizan) [reply src/app/layout.tsx:18]: I am using useState in it so I need to use 'useclient' at top and when we use this we get error when we try to export something

File: src/utils/theme.ts (hunk 2 above)
Comment (ItsFaizan) [reply src/utils/theme.ts:30]: It is very visible with these color when I move to dark mode that I have shifted

**Pull request merged**